### PR TITLE
add section to agent overview with user details

### PR DESCRIPTION
**Pull request by StarfallProjects**

### CODE REVIEW

File: pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb [hunk 1]
```diff
@@ -215,3 +215,7 @@ configuration has a *License type* of `Cores`.
 Ensure your AMI has been [configured to auto-login as the `ec2-user`](#step-2-build-an-ami)
 in the GUI. To allow your pipelines to use Xcode and the iOS Simulator the
 Buildkite Agent launchd job configuration requires an `Aqua` session type.
+
+### What user does the agent run as?
+
+The Buildkite agent runs as ec2-user.
```
Comment (plaindocs) [inline pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb:221]: Is this `ec2-user` or is this "the EC2 user"?

File: pages/agent/v3/linux.md.erb [hunk 2]
```diff
@@ -42,6 +42,10 @@ See the [Agent SSH Keys](/docs/agent/v3/ssh-keys) documentation for more details
 
 The configuration file is located at `~/.buildkite-agent/buildkite-agent.cfg`. See the [configuration documentation](/docs/agent/v3/configuration) for an explanation of each configuration setting.
 
+## Which user the agent runs as
+
+On a generic Linux installation, all commands run as the invoking user.
```
Comment (plaindocs) [inline pages/agent/v3/linux.md.erb:47]: I'm not super keen on "invoking user" but can't quite pin down a better alternative.

```suggestion
On Linux installation, all commands run as the logged in user.
```

File: pages/agent/v3/macos.md.erb [hunk 3]
```diff
@@ -50,6 +50,10 @@ Linux installer script file locations:
 
 See the [configuration documentation](/docs/agent/v3/configuration) for an explanation of each configuration setting.
 
+## Which user the agent runs as
+
+On macOS, the Buildkite agent runs as whichever user the launched service configuration is loaded as.
```
Comment (plaindocs) [inline pages/agent/v3/macos.md.erb:55]: How about?

```suggestion
On macOS, the Buildkite agent runs as the user who started the launched service.
```

File: pages/agent/v3/linux.md.erb (hunk 2 above)
Comment (plaindocs) [reply pages/agent/v3/linux.md.erb:47]: Ah, I see the problem. On a non red hat linux..... ?

**Review by plaindocs — CHANGES_REQUESTED** [review]

File: pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb (hunk 1 above)
Comment (StarfallProjects) [reply pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb:221]: According to Keith's notes, "Elastic Mac for AWS, the ec2-user"

Comment (StarfallProjects) [reply pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb:221]: A quick google suggests there actually is something called the ec2-user. E.g https://stackoverflow.com/a/60029049/2291838

File: pages/agent/v3/linux.md.erb (hunk 2 above)
Comment (StarfallProjects) [reply pages/agent/v3/linux.md.erb:47]: So, the logged in user might invoke commands as at least two different users, maybe? normal and root/sudo? (my shaky knowledge of Linux may be showing)

File: pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb (hunk 1 above)
Comment (plaindocs) [reply pages/agent/v3/elastic_ci_stack_for_ec2_mac/autoscaling_mac_metal.md.erb:221]:
```suggestion
The Buildkite agent runs as `ec2-user`.
```

File: pages/agent/v3/linux.md.erb (hunk 2 above)
Comment (plaindocs) [reply pages/agent/v3/linux.md.erb:47]: :+1:

**Review by plaindocs — APPROVED** [review]

**Pull request merged**